singer_sdk/target_base.py

```diff
@@ -617,3 +635,114 @@ def _merge_missing(source_jsonschema: dict, target_jsonschema: dict) -> None:
         super().append_builtin_config(config_jsonschema)
 
     pass
+
+    @final
+    def add_sqlsink(
```

Comment:
Wdyt about renaming this to `register_sql_sink`?

Reply:
I like the name `register_sql_sink`, but I don't know about renaming the method.  Since I copied `add_sink` it made sense to be `add_sqlsink` so the names were close enough you would know `add_sink` was the original source since we couldn't overload it.  I would be total ok with changing `add_sink` to `register_sink` and `add_sqlsink` to `register_sql_sink`. Wdyt

Reply:
Hmm, yeah it should be safe to rename `add_sink` but even though folks are not supposed to override it, they might still be overriding a different method that calls it. We could leave it as an alias of `register_sink` and add a deprecation warning but that seems unnecessarily complicated.

Let's just leave `add_sqlsink` and `add_sink` as they are.